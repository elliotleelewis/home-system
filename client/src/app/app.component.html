<div class="container mx-auto p-12">
	<div class="grid grid-cols-4 gap-4">
		<div class="flex flex-col">
			<a [routerLink]="'/'">
				<h1 class="p-4 text-center text-3xl font-bold">Home System</h1>
			</a>
			<ul class="menu rounded-box bg-base-200">
				<li>
					<p class="menu-title">
						<i class="fas fa-tools mr-1"></i>
						ShopVac
					</p>
					<ul>
						<li>
							<a
								[routerLink]="'/shop-vac/blast-gates'"
								[routerLinkActive]="'active'"
								>Blast Gates</a
							>
						</li>
					</ul>
				</li>
			</ul>
		</div>
		<div class="col-span-3">
			<router-outlet></router-outlet>
		</div>
	</div>
</div>
