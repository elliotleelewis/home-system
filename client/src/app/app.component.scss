// Imports
@import 'bulma/sass/utilities/all';

// Styles
:host {
	min-block-size: 100vh;
}
