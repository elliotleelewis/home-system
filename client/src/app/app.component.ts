import { ChangeDetectionStrategy, Component } from '@angular/core';
import { RouterLink, RouterLinkActive, RouterOutlet } from '@angular/router';

@Component({
	selector: 'app-root',
	templateUrl: './app.component.html',
	changeDetection: ChangeDetectionStrategy.OnPush,
	standalone: true,
	imports: [RouterLink, RouterLinkActive, RouterOutlet],
})
export class AppComponent {}
