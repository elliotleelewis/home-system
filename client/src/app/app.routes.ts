import { type Routes } from '@angular/router';

import { HomeComponent } from './pages/home/home.component';
import { ShopVacBlastGatesComponent } from './pages/shop-vac-blast-gates/shop-vac-blast-gates.component';

export const ROUTES: Routes = [
	{ path: '', component: HomeComponent },
	{
		path: 'shop-vac/blast-gates',
		component: ShopVacBlastGatesComponent,
	},
];
