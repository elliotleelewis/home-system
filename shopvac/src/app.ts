import express from 'express';
import bodyParser from 'body-parser';

const app = express();

app.set('port', '8080');
app.use(bodyParser.json());
app.use(bodyParser.urlencoded({ extended: true }));

app.get(
	'/',
	(req: express.Request, res: express.Response): void => {
		res.send({ test: 'hello world!' });
	},
);

export default app;
