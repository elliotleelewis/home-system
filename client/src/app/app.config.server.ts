import { mergeApplicationConfig } from '@angular/core';
import type { ApplicationConfig } from '@angular/core';
import { provideServerRendering } from '@angular/ssr';

import { APP_CONFIG } from './app.config';

const serverConfig: ApplicationConfig = {
	providers: [provideServerRendering()],
};

export const CONFIG = mergeApplicationConfig(APP_CONFIG, serverConfig);
