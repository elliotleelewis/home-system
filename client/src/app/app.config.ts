import { provideHttpClient } from '@angular/common/http';
import { inject, provideZoneChangeDetection } from '@angular/core';
import type { ApplicationConfig } from '@angular/core';
import {
	provideClientHydration,
	withEventReplay,
} from '@angular/platform-browser';
import { provideRouter } from '@angular/router';
import { InMemoryCache } from '@apollo/client/core';
import { provideApollo } from 'apollo-angular';
import { HttpLink } from 'apollo-angular/http';

import { ENVIRONMENT } from '../environments/environment';

import { ROUTES } from './app.routes';

export const APP_CONFIG: ApplicationConfig = {
	providers: [
		provideZoneChangeDetection({ eventCoalescing: true }),
		provideApollo(() => {
			const httpLink = inject(HttpLink);
			return {
				cache: new InMemoryCache(),
				link: httpLink.create({
					uri: `${ENVIRONMENT.serverUrl}/graphql`,
				}),
				devtools: {
					enabled: !ENVIRONMENT.production,
				},
			};
		}),
		provideHttpClient(),
		provideRouter(ROUTES),
		provideClientHydration(withEventReplay()),
	],
};
