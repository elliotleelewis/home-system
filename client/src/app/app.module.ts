import { HttpClientModule } from '@angular/common/http';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { BrowserModule } from '@angular/platform-browser';
import { RouterModule } from '@angular/router';
import { InMemoryCache } from '@apollo/client';
import { APOLLO_OPTIONS } from 'apollo-angular';
import { HttpLink } from 'apollo-angular/http';

import { ENVIRONMENT } from '../environments/environment';

import { AppComponent } from './app.component';
import { HomeComponent } from './pages/home/home.component';
import { ShopVacBlastGatesComponent } from './pages/shop-vac-blast-gates/shop-vac-blast-gates.component';

@NgModule({
	declarations: [AppComponent, HomeComponent, ShopVacBlastGatesComponent],
	imports: [
		HttpClientModule,
		FormsModule,
		BrowserModule,
		RouterModule.forRoot(
			[
				{ path: '', component: HomeComponent },
				{
					path: 'shop-vac/blast-gates',
					component: ShopVacBlastGatesComponent,
				},
			],
			{
				useHash: true,
			},
		),
	],
	providers: [
		{
			provide: APOLLO_OPTIONS,
			useFactory: (httpLink: HttpLink) => ({
				cache: new InMemoryCache(),
				link: httpLink.create({
					uri: `${ENVIRONMENT.serverUrl}/graphql`,
				}),
				connectToDevTools: !ENVIRONMENT.production,
			}),
			deps: [HttpLink],
		},
	],
	bootstrap: [AppComponent],
})
export class AppModule {}
