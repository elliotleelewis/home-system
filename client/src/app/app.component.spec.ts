import type { ComponentFixture } from '@angular/core/testing';
import { provideRouter } from '@angular/router';
import { MockBuilder, MockRender } from 'ng-mocks';

import { AppComponent } from './app.component';

describe('AppComponent', () => {
	let component: AppComponent;
	let fixture: ComponentFixture<AppComponent>;

	beforeEach(() => {
		return MockBuilder(AppComponent).provide(provideRouter([]));
	});

	beforeEach(() => {
		fixture = MockRender(AppComponent);
		component = fixture.componentInstance;
	});

	it('should create the app', () => {
		expect(component).toBeTruthy();
	});

	it('should render', () => {
		expect(fixture).toMatchSnapshot();
	});
});
