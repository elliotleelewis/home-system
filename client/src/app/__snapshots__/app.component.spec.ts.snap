// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render 1`] = `
<app-root
  __ngContext__={[Function Number]}
>
  <section
    class="section"
  >
    <div
      class="container"
    >
      <div
        class="columns"
      >
        <div
          class="column is-one-quarter"
        >
          <h1
            class="title is-1 has-text-centered"
          >
            <a
              class="has-text-dark"
              href="/"
            >
              Home System
            </a>
          </h1>
          <div
            class="box"
          >
            <aside
              class="menu has-background-white"
            >
              <p
                class="menu-label"
              >
                <span
                  class="icon"
                >
                  <i
                    class="fas fa-tools"
                  />
                </span>
                 ShopVac 
              </p>
              <ul
                class="menu-list"
              >
                <li>
                  <a
                    href="/shop-vac/blast-gates"
                  >
                    Blast Gates
                  </a>
                </li>
              </ul>
            </aside>
          </div>
        </div>
        <div
          class="column is-three-quarters"
        >
          <router-outlet />
        </div>
      </div>
    </div>
  </section>
</app-root>
`;
